### ISSUE TITLE
Multi-View VAE with Spatial Concatenation Leads to Blurry Motion and Inter-View Inconsistency

### ISSUE THREAD

**Issue by supeng0924** [open-gigaai/giga-world-0#13]
Hi, I’m extending the giga-world codebase to support three camera views (top, left, right) by spatially concatenating their VAE latents along the width dimension. Below is the core part of my modification during training:

       # 1. 获取三个视角的图像
        images_top = batch_dict['images_top']
        images_left = batch_dict['images_left']
        images_right = batch_dict['images_right']
        
        prompt_embeds = batch_dict['prompt_embeds']
        batch_size = images_top.shape[0]
        fps = batch_dict['fps'][0]

        # 2. 分别进行 VAE 编码
        latents_top = self.forward_vae(images_top)
        latents_left = self.forward_vae(images_left)
        latents_right = self.forward_vae(images_right)

        # 3. 在宽度方向拼接 (B, C, T, H, W)，W 是最后一个维度 (dim=4)
        # 拼接顺序建议: Left, Top, Right
        latents = torch.cat([latents_left, latents_top, latents_right], dim=4)

        # 4. 处理 Padding Mask (需要适配拼接后的宽度)
        # 原始单个视角的宽高
        h, w = images_top.shape[-2], images_top.shape[-1]
        # 拼接后的总宽度是 3 * w
        padding_mask = torch.zeros((batch_size, 1, h, w * 3), dtype=self.dtype, device=self.device)

        # 5. 加噪
        input_latents, timesteps = self.edm_loss.add_noise(latents)

        # 6. 处理参考图像 (同样分别编码后拼接)
        ref_images_top = batch_dict['ref_images_top']
        ref_images_left = batch_dict['ref_images_left']
        ref_images_right = batch_dict['ref_images_right']
        ref_masks = batch_dict['ref_masks'].to(self.dtype) # Mask 是共享的

        ref_latents_top = self.forward_vae(ref_images_top)
        ref_latents_left = self.forward_vae(ref_images_left)
        ref_latents_right = self.forward_vae(ref_images_right)
        
        # 同样顺序拼接参考 Latents
        ref_latents = torch.cat([ref_latents_left, ref_latents_top, ref_latents_right], dim=4)

However, the generated videos show two major issues:
Severe blurriness during motion — especially noticeable on the robotic arm when it moves, while static scenes look acceptable.
Inconsistency between views —  the main (top) view and wrist often show misaligned poses or timing.
My final training loss stabilizes around 0.05.

I have two specific questions:
Could spatial concatenation of independently encoded views be fundamentally limiting temporal and cross-view consistency? Since each view is encoded in isolation, the model may lack explicit constraints to align motion or semantics across views.
For full training from scratch with this 3-view setup, what range of final loss would be considered reasonable or expected? Is 0.05 too high, or within normal bounds?

Thank you for your time and this excellent codebase!

**Comment by 1zeryu** [comment]
Apologies for the late reply. May I have a look at the specific effect first? Our model has not been pre-trained in multiple perspectives, which might lead to poor performance. You can perform similar fine-tuning on wan2.2-5b.
